api/tests/unit/infrastructure/serializers/csv/schooling-registration-parser_test.js

```diff
@@ -91,20 +91,95 @@ describe('Unit | Infrastructure | SchoolingRegistrationParser', () => {
             status: 'AP',
           });
         });
+
+        it('should not return error given nationalIdentifier with different status', () => {
+          const input = `${schoolingRegistrationCsvColumns}
+          123F;Beatrix;The;Bride;Kiddo;Black Mamba;01/01/1970;97422;;974;99100;AP;MEF1;Division 1;
+          123F;Beatrix;The;Bride;Kiddo;Black Mamba;01/01/1970;97422;;974;99100;ST;MEF1;Division 1;
+          `;
+          const organizationId = 789;
+          const encodedInput = iconv.encode(input, 'utf8');
+          const parser = new SchoolingRegistrationParser(encodedInput, organizationId, true);
+  
+          const { registrations } = parser.parse();
+          
+          expect(registrations[0]).to.includes({
+            nationalStudentId: undefined,
+            nationalApprenticeId: '123F',
+            status: 'AP',
+          });
+
+          expect(registrations[1]).to.includes({
+            nationalStudentId: '123F',
+            nationalApprenticeId: undefined,
+            status: 'ST',
+          });
+        });
       });
 
       context('when the data are not correct', () => {
-        it('should throw an EntityValidationError', async () => {
+        it('should throw an EntityValidationError with malformated birthDate', async () => {
+          //given
           const input = `${schoolingRegistrationCsvColumns}
-      123F;Beatrix;The;Bride;Kiddo;Black Mamba;aaaaa;97422;;200;99100;ST;MEF1;Division 1;
-      `;
+          123F;Beatrix;The;Bride;Kiddo;Black Mamba;aaaaa;97422;;200;99100;ST;MEF1;Division 1;
+          `;
           const encodedInput = iconv.encode(input, 'utf8');
           const parser = new SchoolingRegistrationParser(encodedInput, 123);
 
           const error = await catchErr(parser.parse, parser)();
-
+          
+          //then
           expect(error).to.be.an.instanceOf(EntityValidationError);
         });
+
+        context('when organization is SCO and CSV has duplicates on national identifier' , () => {
+          it('should throw an CsvImportError', async () => {
+            const input = `${schoolingRegistrationCsvColumns}
+            123F;Beatrix;The;Bride;Kiddo;Black Mamba;01/05/1986;97422;;200;99100;ST;MEF1;Division 1;
+            123F;Beatrix;The;Bride;Kiddo;Black Mamba;01/05/1986;97422;;200;99100;AP;MEF1;Division 1;
+            `;
+
+            const encodedInput = iconv.encode(input, 'utf8');
+            const parser = new SchoolingRegistrationParser(encodedInput, 123);
+  
+            const error = await catchErr(parser.parse, parser)();
+            
+            //then
+            expect(error).to.be.an.instanceOf(CsvImportError);
+          });
+        });
+
+        context('when organization is SCO Agriculture and CSV has duplicates on national identifier' , () => {
+          it('should throw an CsvImportError, with all status AP', async () => {
```

Comment:
**question (non-blocking) :** Je me demande si on ne peut pas garder qu'un des deux tests ? Je ne voit pas trop ce que le tests met en évidence ?
J'ai l'impression que peu importe le status si le `nationalIdentifier` est le même l'erreur est lancée, je dirais que si le status n'a pas d'impact dans le test alors un seul test suffit

Reply:
en fait il manque un test . si il est Agri AP ST avec le même identifiant unique. ça ne doit pas casser .

il manquerait un should not throw error while nationalIdentifier same with different status.

Reply:
En revanche ça doit casser pour les non agri.